Remove brownie

calina-c opened the pull request:
Closes [#1460](https://github.com/oceanprotocol/ocean.py/issues/1460) "Replace brownie with web3.py"
Closes #1451 (build tests failing due to dependency issues)

trentmc commented on READMEs/install.md:
```diff
@@ -55,17 +55,9 @@ Issue: MacOS "Unsupported Architecture"
 
 - ocean.py has two dependencies that don't support Python 3.11 yet. Here's the workaround for each.
-- (1) The `eth-brownie` package uses Vyper which doesn't support Python 3.11. The workaround: before installing ocean-lib, run `pip install vyper==0.3.7 --ignore-requires-python` and `sudo apt-get install python3.11-dev`
+- (1) Vyper which doesn't support Python 3.11. The workaround: before installing ocean-lib, run `pip install vyper==0.3.7 --ignore-requires-python` and `sudo apt-get install python3.11-dev`
```
Why was this line kept at all? Afaik, it was only Brownie that uses Vyper. If that's correct, we can remove the line.

calina-c replied:
You are right, I removed it.

trentmc commented on ocean_lib/services/test/test_service.py:
```diff
@@ -155,10 +154,10 @@ def test_utilitary_functions_for_trusted_algorithm_publishers(publisher_ocean):
     assert compute_service.type == "compute"
 
-    addr1 = accounts.add().address
+    addr1 = publisher_ocean.config_dict["web3_instance"].eth.account.create().address
```
`publisher_ocean.config_dict["web3_instance"]` is repeated 3 times in a row. Hard to grok, and verbose.

I recommend:
```python
web3 = publisher_ocean.config_dict["web3_instance"]
addr1 = web3.eth.account.create().address() 
..
addr2 = web3.eth.account.create().address() 
..
addr2 = web3.eth.account.create().address() 
```

calina-c replied:
fixed

trentmc reviewed (changes requested):
In general, this looks great! Some super cool things. Especially the smart contract wrapper, so that our py objects behave like deployed contracts just like brownie, without a bunch of manually-written code.

I made a couple tweaks.

TODOs

1. I left just a couple comments in the code, should be easy to fix.

2. it seems that two mumbai-related tests are randomly failing. It'd be good to get to the bottom of this before merging.
- `tests/integration/remote/test_mumbai_main.py::test_ocean_tx__create`
- `tests/integration/remote/test_mumbai_readme.py::test_simple_remote_readme`

TODOs after merged:

3. I found that custom gas strategy fails with `create_url_asset()`. I created a new issue for that [#1464](https://github.com/oceanprotocol/ocean.py/issues/1464) 
4. As the "remove brownie" issue itself [#1460](https://github.com/oceanprotocol/ocean.py/issues/1460) points out, once this is merged then there may be other github issues that this has resolved; we can check to see if they have, and close accordingly.

calina-c commented:
> In general, this looks great! Some super cool things. Especially the smart contract wrapper, so that our py objects behave like deployed contracts just like brownie, without a bunch of manually-written code.

Thanks! I did focus on keeping most of the interface as it was. Most changes are actually inside tests or inside combo functions that followed up on triggered events and other such artifacts. It should be super easy for users to switch to this new version once it's released.

> TODOs
> 
> 1. I left just a couple comments in the code, should be easy to fix.
Fixed.

> 2. it seems that two mumbai-related tests are sporadically failing, [see here](https://github.com/oceanprotocol/ocean.py/actions/workflows/pytest.yml). It'd be good to get to the bottom of this before merging.
>    
>    * `tests/integration/remote/test_mumbai_main.py::test_ocean_tx__create`
>    * `tests/integration/remote/test_mumbai_readme.py::test_simple_remote_readme`
Fixed. It seems I was too strict with the skippable error code for cap. It seems to be working all right now.

> TODOs after merged:
> 
> 3. I found that custom gas strategy fails with `create_url_asset()`. I created a new issue for that [#1464](https://github.com/oceanprotocol/ocean.py/issues/1464)
> 4. As the "remove brownie" issue itself [#1460](https://github.com/oceanprotocol/ocean.py/issues/1460) points out, once this is merged then there may be other github issues that this has resolved; we can check to see if they have, and close accordingly.
I think I should address no. 3 before releasing the new version. What do you recommend? After merging, but before releasing is what I think.

trentmc commented:
> I think I should address no. 3 before releasing the new version. What do you recommend? After merging, but before releasing is what I think.

Yes.

Same for 4.

trentmc commented:
Closes https://github.com/oceanprotocol/ocean.py/issues/1460 "Replace brownie with web3.py"

Closes https://github.com/oceanprotocol/ocean.py/issues/1451 (build tests failing due to dependency issues)
